apps/src/templates/projects/SectionProjectsList.jsx

```diff
@@ -0,0 +1,89 @@
+import React, {Component, PropTypes} from 'react';
+import ProjectsList from './ProjectsList';
+import StudentFilterDropdown from './StudentFilterDropdown';
+import _ from 'lodash';
+import color from "../../util/color";
+
+const ALL_STUDENTS = '_all_students';
+
+const styles = {
+  filterComponent: {
+    float: 'right',
+  },
+  filterRow: {
+    backgroundColor: color.teal,
+    borderBottom: 'solid 1px white',
+    padding: 10,
+  },
+  clearDiv: {
+    clear: 'both'
+  }
+};
+
+class SectionProjectsList extends Component {
+  constructor(props) {
+    super(props);
+
+    const studentNames = SectionProjectsList.getStudentNames(props.projectsData);
+
+    this.state = {
+      studentNames,
+      selectedStudent: ALL_STUDENTS,
+    };
+  }
+
+  onChangeStudent(selectedStudent) {
+    this.setState({selectedStudent});
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const studentNames = SectionProjectsList.getStudentNames(nextProps.projectsData);
+    let newState = {studentNames};
```

Comment:
Non-blocking nit: Could be `const`, which merely asserts that `newState` will always refer to this object, not that the object will never change.